File: capture_screenshots.py
import os
import time
import subprocess
from selenium import webdriver
from selenium.webdriver.chrome.options import Options
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
import base64

# Setup Chrome options for headless
chrome_options = Options()
chrome_options.add_argument("--headless")
chrome_options.add_argument("--disable-gpu")
chrome_options.add_argument("--window-size=400,900")
chrome_options.add_argument("--hide-scrollbars")
chrome_options.add_argument("--force-device-scale-factor=1")

# Screens to capture
screens = [
    {"name": "01_home", "click": None, "wait": 3},
    {
        "name": "02_translate",
        "click": ("xpath", "//button[contains(@onclick, 'translate')]"),
        "wait": 5,
    },
    {
        "name": "03_learn",
        "click": ("xpath", "//button[contains(@onclick, 'learn')]"),
        "wait": 3,
    },
    {
        "name": "04_settings",
        "click": ("xpath", "//button[contains(@onclick, 'settings')]"),
        "wait": 3,
    },
]

output_dir = r"C:\Users\A2Z\OneDrive\Documents\NDN Analytics\Hackaton\Rais 26 by Dorahacks\assets\screenshots"
os.makedirs(output_dir, exist_ok=True)

try:
    driver = webdriver.Chrome(options=chrome_options)
    driver.get("https://sign-interpreter-five.vercel.app")

    time.sleep(5)  # Wait for page load

    # Take initial screenshot
    for i, screen in enumerate(screens):
        print(f"Capturing {screen['name']}...")

        # Click to navigate if needed
        if screen.get("click"):
            click_type, click_value = screen["click"]
            elem = driver.find_element(By.XPATH, click_value)
            elem.click()

        time.sleep(screen["wait"])

        # Save screenshot
        filepath = os.path.join(output_dir, f"{screen['name']}.png")
        driver.save_screenshot(filepath)
        print(f"Saved: {filepath}")

    print("All screenshots captured!")

except Exception as e:
    print(f"Error: {e}")
finally:
    driver.quit()

File: capture_screenshots.js
const { chromium } = require('playwright');

(async () => {
    const browser = await chromium.launch({ headless: true });
    const context = await browser.newContext({
        viewport: { width: 375, height: 812 } // Mobile viewport
    });
    const page = await context.newPage();
    
    // Listen for console errors
    page.on('console', msg => {
        if (msg.type() === 'error') console.log('Console Error:', msg.text());
    });

    const baseUrl = 'https://sign-interpreter-five.vercel.app';
    const screenshots = [
        { name: '01_home', url: baseUrl },
        { name: '02_translate', url: `${baseUrl}#translate`, click: '#translate-screen button:first-child' },
        { name: '03_learn', url: `${baseUrl}#learn` },
        { name: '04_numbers', url: `${baseUrl}#numbers` },
        { name: '05_words', url: `${baseUrl}#words` },
        { name: '06_face', url: `${baseUrl}#face` },
        { name: '07_history', url: `${baseUrl}#history` },
        { name: '08_settings', url: `${baseUrl}#settings` },
        { name: '09_voice', url: `${baseUrl}#voice-settings` },
        { name: '10_about', url: `${baseUrl}#about` },
    ];

    console.log('Starting screenshot capture...');

    for (const shot of screenshots) {
        try {
            console.log(`Capturing ${shot.name}...`);
            await page.goto(shot.url, { waitUntil: 'networkidle', timeout: 30000 });
            await page.waitForTimeout(2000);
            
            // Skip loading screen
            const loading = await page.$('#loading-screen');
            if (loading) {
                await page.waitForTimeout(3000);
            }
            
            await page.screenshot({ 
                path: `assets/screenshots/${shot.name}.png`,
                fullPage: false 
            });
            console.log(`✓ ${shot.name} captured`);
        } catch (e) {
            console.error(`✗ ${shot.name} failed:`, e.message);
        }
    }

    await browser.close();
    console.log('Done!');
})();
